src/source/file/file_server.rs

```diff
@@ -56,15 +55,106 @@ impl Default for FileServerConfig {
     }
 }
 
-impl FileServer {
-    /// Make a FileServer
-    ///
-    pub fn new(chans: util::Channel, config: FileServerConfig) -> FileServer {
-        FileServer {
-            chans: chans,
-            pattern: config.path.expect("must specify a 'path' for FileServer"),
-            tags: config.tags,
-            max_lines_read: config.max_lines_read,
+fn file_server(mut chans: util::Channel, config: FileServerConfig, poller: mio::Poll) {
```

Comment:
This... makes me nervous. Why are we getting rid of trait implementations here?

Reply:
New have been moved into the Source trait.

Reply:
Okay. I guess what I'm getting at is I'd like to see `file_server` as part of some standard interface. Maybe a `handle` on Source?

Reply:
Ja, let's continue to discuss `handle` in internal.rs's thread.